src/repositories/ItemRateRepository.ts

```diff
@@ -0,0 +1,70 @@
+import DBService, {
+  DBLevel,
+  DefaultOrDateItemKey,
+  FacilityIndexKey,
+  FacilityItemValues,
+  FacilityValues,
+  FormattedDate,
+  LevelDefaultTyping
+} from '../services/DBService';
+import ApiError from '../exceptions/ApiError';
+import { AbstractSublevel } from 'abstract-level';
+import { Rates } from '../proto/lpms';
+
+
+abstract class ItemRateRepository {
+  protected dbService = DBService.getInstance();
+  protected db: AbstractSublevel<AbstractSublevel<AbstractSublevel<DBLevel, LevelDefaultTyping, string, FacilityValues>,
+    LevelDefaultTyping,
+    string,
+    FacilityItemValues>,
+    LevelDefaultTyping,
+    DefaultOrDateItemKey,
+    Rates>;
+
+  protected constructor(
+    facilityId: string,
+    indexKey: FacilityIndexKey,
+    itemId: string
+  ) {
+    this.db = this.dbService.getItemRatesDB(facilityId, indexKey, itemId);
+  }
+
+  public async getRate(key: DefaultOrDateItemKey): Promise<Rates | null> {
+    try {
+      return await this.db.get(key);
+    } catch (e) {
+      if (e.status === 404) {
+        throw ApiError.NotFound(`Unable to get "${key}" of rate level"`);
+      }
+      throw e;
+    }
+  }
+
+  public async setRateByDate(
+    key: FormattedDate,
+    value: Rates
+  ): Promise<void> {
+    await this.db.put(key, value);
+  }
+
+  public async setDefaultRate(value: Rates): Promise<void> {
```

Comment:
Suggest that we try to keep to the same syntax, like `setRate`, `setRateDefault`, `delRate` etc.